systemtest/src/main/java/io/strimzi/systemtest/logs/LogCollector.java

```diff
@@ -68,22 +70,32 @@ public LogCollector(String testSuiteName, String testCaseName, KubeClient kubeCl
 
         this.testSuite = new File(logSuiteDir);
 
-        // @BeforeAll in AbstractST always pass, and I ensure that we always deploy CO and after that some error can occur
-        this.clusterOperatorPod = kubeClient.getClient().pods().inAnyNamespace().list().getItems().stream()
-            .filter(pod -> pod.getMetadata().getName().contains(Constants.STRIMZI_DEPLOYMENT_NAME))
+        Optional<Pod> optionalCoPod = kubeClient.getClient().pods().inAnyNamespace().list().getItems().stream()
+            .filter(pod -> pod.getMetadata().getName().contains(ResourceManager.getCoDeploymentName()))
             // contract only one Cluster Operator deployment inside all namespaces
-            .findFirst()
-            .orElseThrow();
-        this.clusterOperatorNamespace = this.clusterOperatorPod.getMetadata().getNamespace();
+            .findFirst();
+
+        if (optionalCoPod.isPresent()) {
+            // @BeforeAll in AbstractST always pass, and I ensure that we always deploy CO and after that some error can occur
+            this.clusterOperatorPod = optionalCoPod.get();
+            this.clusterOperatorNamespace = this.clusterOperatorPod.getMetadata().getNamespace();
+        }
+
         this.testCase = new File(logSuiteDir + "/" + testCaseName);
 
         boolean logDirExist = this.logDir.exists() || this.logDir.mkdirs();
         boolean logTestSuiteDirExist = this.testSuite.exists() || this.testSuite.mkdirs();
         boolean logTestCaseDirExist = this.testCase.exists() || this.testCase.mkdirs();
 
-        if (!logDirExist) throw new IOException("Unable to create path");
-        if (!logTestSuiteDirExist) throw new IOException("Unable to create path");
-        if (!logTestCaseDirExist) throw new IOException("Unable to create path");
+        if (!logDirExist) {
+            throw new IOException("Unable to create path");
+        }
+        if (!logTestSuiteDirExist) {
+            throw new IOException("Unable to create path");
+        }
+        if (!logTestCaseDirExist) {
+            throw new IOException("Unable to create path");
+        }
```

Comment:
is this needed? The code has the same semantics but If you prefer this style I am ok with it...

Reply:
I don't like one-line ifs. From my POV it's quite messy. Maybe we can ask others what they think about it.